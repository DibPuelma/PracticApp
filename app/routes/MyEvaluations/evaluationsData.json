{
  "evaluations": [
    {
      "average": 3.7,
      "comment": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
      "store": "Pr√ºne",
      "logo": "https://www.republicamoda.com/media/catalog/category/20120302-logoPrune_1.jpg"
    },
    {
      "average": 2.5,
      "comment": "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.",
      "store": "Reebok",
      "logo": "http://static.dnaindia.com/sites/default/files/2015/04/19/328875-reebok-logo.jpg"
    },
    {
      "average": 5.0,
      "comment": "Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
      "store": "Amphora",
      "logo": "http://zapalook.com.ar/wp-content/uploads/2015/07/logo-Amphora.jpg"
    },
    {
      "average": 4.2,
      "comment": "Meeeeeeeeh",
      "store": "Microplay",
      "logo": "http://www.tutarjeta.cl/wp-content/uploads/2016/01/logo-microplay.jpg"
    }

  ]
}
