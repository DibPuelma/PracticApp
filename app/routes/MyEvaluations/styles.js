import {
  StyleSheet
} from 'react-native'

module.exports = StyleSheet.create({
  container: {
    flexWrap: 'wrap',
    justifyContent: ''
  },
  image: {
    width: 60,
    height: 60
  },
  comment: {

  }
})
