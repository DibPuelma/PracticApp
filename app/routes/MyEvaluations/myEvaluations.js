import React, { Component } from 'react';
import {
  View,
  ListView,
  Text,
  Image,
  TouchableHighlight,
  ActivityIndicator
} from 'react-native';

import evaluationsData from './evaluationsData';
import styles from './styles';
import backButtonHandler from '../../lib/backButtonHandler';
import LoadingSpinner from '../../components/LoadingSpinner/loadingSpinner';

export default class MyEvaluations extends Component{
  constructor(props) {
    super(props);
    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
    this.state = {
      dataSource: ds.cloneWithRows(["row 1", "row 2"]),
      ready: false,
      uri: 'https://practicapi.herokuapp.com/user/' + props.user.id + '/answered_poll'
    };
    this._backToPrevious = this._backToPrevious.bind(this);
    this.singletonBackButtonHandler = backButtonHandler.getInstance();
  }

  componentDidMount(){
    fetch(this.state.uri, {
      method: 'GET',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
      }
    })
    .then((response) => response.json())
    .then((responseJson) => {
      this.setState({ dataSource: this.state.dataSource.cloneWithRows(responseJson)});
      this.setState({ ready: true});
    })
    .catch((error) => {
      console.error(error);
    });
  }

  componentWillMount(){

    this.singletonBackButtonHandler.addBackEvent(this._backToPrevious);
  }

  componentWillUnmount() {
    this.singletonBackButtonHandler.removeBackEvent(this._backToPrevious);
  }

  _backToPrevious() {
    this.props.navigator.replace({id:'QRReader'});
    return true; // This is important to prevent multiple calls
  }
  //TODO: add logo
  render() {
    if (!this.state.ready) {
      return (
        <LoadingSpinner/>
      );
    }
    else {
      return (
        <View style={styles.container}>
        <ListView
        dataSource={this.state.dataSource}
        renderRow={(rowData) => (
          <TouchableHighlight onPress={() => this._goToDetails(rowData.id, rowData.SellPoint !== null ? rowData.SellPoint.location : "Sin tienda",rowData.SellPoint !== null ? rowData.SellPoint.Company.logo : "http://www.herorewardsclub.com/images/default-merchant-logo.png")} >
          <View style={styles.listElement}>
          <Image source={{uri: rowData.SellPoint !== null ? rowData.SellPoint.Company.logo : "http://www.herorewardsclub.com/images/default-merchant-logo.png"}} style={styles.image} />
          {this._getComment(rowData.Answers)}

          <Text style={styles.average}>{rowData.id}</Text>
          </View>
          </TouchableHighlight>
        )
      }
      />
      </View>
    );
  }

}
_getComment(answers){
  var comment = "No se registró ningún comentario"
  answers.map((data) => {
    if(data.string_value !== null){
      comment = data.string_value;
    }
  })
  return (<Text numberOfLines={3} style={styles.comment}>{comment}</Text>)
}
_goToDetails(id, name, logo){
  var sellPointData = {
    id:id,
    logo:logo,
    name:name
  };

  this.props.navigator.push({id:'EvaluationDetails', sellPointData:sellPointData})
}
}
