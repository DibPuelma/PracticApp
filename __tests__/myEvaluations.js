import 'react-native';
import React from 'react';
import MyEvaluations from '../app/routes/MyEvaluations/myEvaluations'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <MyEvaluations />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
