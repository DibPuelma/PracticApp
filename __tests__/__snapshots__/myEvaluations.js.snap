exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 72,
    }
  }>
  <ScrollView
    dataSource={
      ListViewDataSource {
        "items": 4,
      }
    }
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    removeClippedSubviews={true}
    renderRow={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#565656",
            "flex": 5,
            "flexDirection": "row",
            "padding": 5,
          }
        }>
        <Image
          source={
            Object {
              "uri": "https://www.republicamoda.com/media/catalog/category/20120302-logoPrune_1.jpg",
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 60,
              "width": 60,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Object {
              "flex": 3,
              "marginLeft": 10,
              "marginRight": 10,
            }
          }>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 30,
              "textAlign": "center",
            }
          }>
          3.7
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#565656",
            "flex": 5,
            "flexDirection": "row",
            "padding": 5,
          }
        }>
        <Image
          source={
            Object {
              "uri": "http://static.dnaindia.com/sites/default/files/2015/04/19/328875-reebok-logo.jpg",
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 60,
              "width": 60,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Object {
              "flex": 3,
              "marginLeft": 10,
              "marginRight": 10,
            }
          }>
          Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 30,
              "textAlign": "center",
            }
          }>
          2.5
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#565656",
            "flex": 5,
            "flexDirection": "row",
            "padding": 5,
          }
        }>
        <Image
          source={
            Object {
              "uri": "http://zapalook.com.ar/wp-content/uploads/2015/07/logo-Amphora.jpg",
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 60,
              "width": 60,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Object {
              "flex": 3,
              "marginLeft": 10,
              "marginRight": 10,
            }
          }>
          Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 30,
              "textAlign": "center",
            }
          }>
          5
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#565656",
            "flex": 5,
            "flexDirection": "row",
            "padding": 5,
          }
        }>
        <Image
          source={
            Object {
              "uri": "http://www.tutarjeta.cl/wp-content/uploads/2016/01/logo-microplay.jpg",
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 60,
              "width": 60,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Object {
              "flex": 3,
              "marginLeft": 10,
              "marginRight": 10,
            }
          }>
          Meeeeeeeeh
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "flex": 1,
              "fontSize": 30,
              "textAlign": "center",
            }
          }>
          4.2
        </Text>
      </View>
    </View>
  </ScrollView>
</View>
`;
